objects/Transaction.go

```diff
@@ -313,9 +319,50 @@ func (t *Transaction) UnmarshalJSON(data []byte) error {
 	return err
 }
 
+type MarshalStruct struct {
+	json.RawMessage
+	objects.Object
+	transactions.Transaction
+}
+
 func (t *Transaction) MarshalJSON() ([]byte, error) {
-	switch t.ObjectType {
+	trsnsactionData, err := t.getTransactionData()
+	if err != nil {
+		return nil, err
+	}
+
+	dataMap := map[string]any{}
+	err = json.Unmarshal(trsnsactionData, &dataMap)
+	if err != nil {
+		return nil, err
+	}
+
+	txWrapper := struct {
```

Comment:
Thoughts on declaring the struct instead of having it be anonymous struct. It's used in both the marshal and unmarshal methods

Reply:
Maybe this is what the `MarshalStruct` was intended for